bcda/auth/client/ssas_test.go

```diff
@@ -105,200 +158,163 @@ func (s *SSASClientTestSuite) TestNewSSASClient_TLSFalseNoURL() {
-		w.WriteHeader(http.StatusCreated)
-		_, err := w.Write([]byte(`{ "ID": 123456 }`))
-		if err != nil {
-			log.Fatal(err)
-		}
-	})
-	server := httptest.NewServer(router)
-
-	conf.SetEnv(s.T(), "SSAS_URL", server.URL)
-	conf.SetEnv(s.T(), "SSAS_USE_TLS", "false")
-
-	client, err := authclient.NewSSASClient()
-	if err != nil {
-		s.FailNow(constants.CreateSsasErr, err.Error())
+func (s *SSASClientTestSuite) TestCreateGroupTable() {
+	tests := []struct {
+		fnInput       []string
+		header        int
+		env           EnvVars
+		errorExpected bool
+		errorMessage  string
+	}{
+		{fnInput: []string{"1", "name", "", `{ "ID": 123456 }`}, header: http.StatusCreated, env: EnvVars{}, errorExpected: false, errorMessage: ""},
+		{fnInput: []string{"1", "name", "", `{ "ID": 123456 }`}, header: http.StatusCreated, env: EnvVars{SSAS_URL: "localhost"}, errorExpected: true, errorMessage: "could not create group: Post \"localhost/group\": unsupported protocol scheme \"\""},
+		{fnInput: []string{"\"", "name", "", `{ "ID": 123456 }`}, header: http.StatusCreated, env: EnvVars{SSAS_URL: "\n"}, errorExpected: true, errorMessage: "invalid input for new group_id \": parse \"\\n/group\": net/url: invalid control character in URL"},
+		{fnInput: []string{"1", "name", "", `{ "ID": 123456 }`}, header: http.StatusCreated, env: EnvVars{BCDA_SSAS_CLIENT_ID: "-1"}, errorExpected: true, errorMessage: "missing clientID or secret"},
+		{fnInput: []string{"", "name", "", `{ "ID": 123456 }`}, env: EnvVars{}, header: http.StatusBadRequest, errorExpected: true, errorMessage: "could not create group: { \"ID\": 123456 }"},
 	}
 
-	resp, err := client.CreateGroup("1", "name", "")
-	if err != nil {
-		s.FailNow("Failed to create group", err.Error())
-	}
+	for _, tc := range tests {
+		router := chi.NewRouter()
+		router.Post("/group", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(tc.header)
+			_, err := w.Write([]byte(tc.fnInput[3]))
+			if err != nil {
+				log.Fatal(err)
+			}
+		})
+		server := httptest.NewServer(router)
 
-	assert.Equal(s.T(), `{ "ID": 123456 }`, string(resp))
-}
+		conf.SetEnv(s.T(), "SSAS_URL", server.URL)
+		conf.SetEnv(s.T(), "BCDA_SSAS_CLIENT_ID", server.URL)
+		s.setEnvVars(tc.env)
+		conf.SetEnv(s.T(), "SSAS_USE_TLS", "false")
 
-func (s *SSASClientTestSuite) TestCreateSystem() {
-	router := chi.NewRouter()
-	router.Post("/system", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusCreated)
-		_, err := w.Write([]byte(`{"system_id": "1", "client_id":` + constants.FakeClientIDBt + `, "client_secret": ` + constants.FakeSecretBt + `, "client_name": "fake-name"}`))
+		client, err := authclient.NewSSASClient()
 		if err != nil {
-			log.Fatal(err)
+			s.FailNow(constants.CreateSsasErr, err.Error())
 		}
-	})
-	server := httptest.NewServer(router)
 
-	conf.SetEnv(s.T(), "SSAS_URL", server.URL)
-	conf.SetEnv(s.T(), "SSAS_USE_TLS", "false")
+		fmt.Println(tc)
```

Comment:
Is this set in case of a test failure/visibility?

Reply:
Good catch, going through and deleting the print statements